format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSink.java

```diff
@@ -64,10 +64,20 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     // invalid
     collector.getOrThrowException();
 
+    if (config.containsMacro(NAME_FORMAT)) {
+      // Deploy all format plugins. This ensures that the required plugin is available when
+      // the format macro is evaluated in prepareRun,
+      for (FileFormat f: FileFormat.values()) {
+        pipelineConfigurer.usePlugin(ValidatingOutputFormat.PLUGIN_TYPE, f.name().toLowerCase(), f.name().toLowerCase(),
```

Comment:
are there any format properties that are required by some formats and optional for others? If so, this can run into errors

Reply:
Thanks for pointing this out. Handled InvalidPluginConfigException as discussed offline.